src/api/index.js

```diff
@@ -258,23 +267,31 @@ export function handleSignIn() {
 // functions to be used by the admin page
 // in order to show data and complete the process of
 // assigning the leader role on protests
-export async function listPendingRequests(uid, phoneNumber) {
-  return firestore.collection('leader_requests').where('status', 'pending').orderBy('created_at', 'desc').limit(20).get();
+export async function listLeaderRequests() {
+  const leaderRequests = [];
+  const snapshot = await firestore
+    .collection('leader_requests')
+    .where('status', '==', 'pending')
+    .orderBy('created_at', 'desc')
+    .limit(20)
+    .get();
+  snapshot.forEach((doc) => {
+    leaderRequests.push({ id: doc.id, ...doc.data() });
+  });
+  return leaderRequests;
 }
 
 // When super-admin approves a protest-user request
-export async function assignRoleOnProtest(userId, protestId, requestId) {
-  await firestore
-    .collection('protests')
-    .doc(protestId)
-    .update({
-      'roles.leaders': firebase.firestore.FieldValue.arrayUnion(userId),
-    });
-
-  // Delete request
-  await firestore.collection('leader_requests').doc(requestId).update({ status: 'done' });
-}
+export async function assignRoleOnProtest({ userId, protestId, requestId, status }) {
+  if (status === 'done') {
```

Comment:
Now when thinking about it, maybe we should go with "approved" and "rejected"? Otherwise we won't be able to look back on it and understand what happened with a certain request.

Reply:
We wanted to replace "done" with approved, but didn't because of backward compatibility. It is an easy change so if all inline, let's do that @ranyitz